proxy/src/control/destination/mod.rs

```diff
@@ -88,11 +88,31 @@ pub struct Resolution<B> {
 
 /// Metadata describing an endpoint.
 #[derive(Clone, Debug, Hash, Eq, PartialEq)]
-struct Metadata {
+pub struct Metadata {
     /// A set of Prometheus metric labels describing the destination.
-    metric_labels: Option<DstLabels>,
+    dst_labels: Option<DstLabels>,
+
+    /// How to verify TLS for the endpoint.
+    tls_identity: Option<Arc<TlsIdentity>>,
+}
+
+/// How to verify TLS for an endpoint.
+///
+/// NOTE: This currently derives `Hash`, `PartialEq`, and `Eq`, which is not
+///       entirely correct, as domain name equality ought to be case
+///       insensitive. However, `Metadata` must be `Hash` + `Eq`, so this is at
+///       least better than having `Metadata` ignore the TLS identity when
+///       checking for equality
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
```

Comment:
Since we always store `TlsIdentity` in an `Arc`, we shouldn't derive `Clone` for it. Similarly, we don't need to derive `Hash` for it any more.after #1058.

We still need PartialEq (and Eq) because `Cache` requires it, IIUC.

Reply:
I've removed the derivation of `Clone` in d2958c93c6575282b266b94adcb92325dc86ef82. I'll remove `Hash` after #1058 lands, either in this branch or in a follow-up, depending on if that branch merges first or not.